Test failure: integrations.sparse.test_simple_fusion_search_integration.TestSearchIntegration

```
$ python -m unittest integrations/sparse/test_simple_fusion_search_integration.py
indexes/lucene-index-cord19-abstract-2020-05-01 already exists, skipping download.
indexes/lucene-index-cord19-full-text-2020-05-01 already exists, skipping download.
indexes/lucene-index-cord19-paragraph-2020-05-01 already exists, skipping download.
Downloading https://git.uwaterloo.ca/jimmylin/covidex-trec-covid-runs/raw/master/round2/anserini.covid-r2.fusion1.txt.gz to runs/anserini.covid-r2.fusion1.txt.gz...
anserini.covid-r2.fusion1.txt.gz: 672kB [00:00, 3.66MB/s]                                                                                                                                                                                                                                               
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 35/35 [06:31<00:00, 11.18s/it]
F
======================================================================
FAIL: test_simple_fusion_searcher (integrations.sparse.test_simple_fusion_search_integration.TestSearchIntegration)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/store/scratch/jimmylin/pyserini/integrations/sparse/test_simple_fusion_search_integration.py", line 64, in test_simple_fusion_searcher
    self.assertTrue(filecmp.cmp('runs/this.txt', 'runs/that.txt'))
AssertionError: False is not true

----------------------------------------------------------------------
Ran 1 test in 393.631s

FAILED (failures=1)
```

Digging in:

```
$ diff runs/this.txt runs/that.txt 
26883,26884c26883,26884
< 27 4lja9w93 883
< 27 9zdqyeaa 884
---
> 27 9zdqyeaa 883
> 27 4lja9w93 884
```

Seems like this is due to a brittle test... I'll think a bit about how to handle.

--- a/integrations/sparse/test_simple_fusion_search_integration.py
+++ b/integrations/sparse/test_simple_fusion_search_integration.py
@@ -56,10 +56,12 @@ class TestSearchIntegration(unittest.TestCase):
         all_topics_run = TrecRun.concat(runs)
         all_topics_run.save_to_txt(output_path='runs/fused.txt', tag='reciprocal_rank_fusion_k=60')
 
-        # Only keep topic, docid and rank. Scores have different floating point precisions.
+        # Only keep topic, docid, and rank. Scores may be slightly different due to floating point precision issues and underlying lib versions.
         # TODO: We should probably do this in Python as opposed to calling out to shell for better portability.
-        os.system("""awk '{print $1" "$3" "$4}' runs/fused.txt > runs/this.txt""")
-        os.system("""awk '{print $1" "$3" "$4}' runs/anserini.covid-r2.fusion1.txt > runs/that.txt""")
+        # This has also proven to be a somewhat brittle test, see https://github.com/castorini/pyserini/issues/947
+        # A stopgap for above issue, we're restricting comparison to only top-100 ranks.
+        os.system("""awk '$4 <= 100 {print $1" "$3" "$4}' runs/fused.txt > runs/this.txt""")
+        os.system("""awk '$4 <= 100 {print $1" "$3" "$4}' runs/anserini.covid-r2.fusion1.txt > runs/that.txt""")
 
         self.assertTrue(filecmp.cmp('runs/this.txt', 'runs/that.txt'))
 

